vm/src/dictdatatype.rs

```diff
@@ -540,23 +577,19 @@ impl<T: Clone> Dict<T> {
             return Ok(None);
         };
         let mut inner = self.write();
-        if matches!(inner.entries.get(entry_index), Some(entry) if entry.index == index_index) {
+        if matches!(inner.entries.get(entry_index), Some(entry) if entry.as_ref().unwrap().index == index_index)
+        {
             // all good
         } else {
             // The dict was changed since we did lookup. Let's try again.
             return Err(());
         };
         inner.indices[index_index] = IndexEntry::DUMMY;
         inner.used -= 1;
-        let removed = if entry_index == inner.used {
-            inner.entries.pop().unwrap()
-        } else {
-            let last_index = inner.entries.last().unwrap().index;
-            let removed = inner.entries.swap_remove(entry_index);
-            inner.indices[last_index] = entry_index as i64;
-            removed
-        };
-        Ok(Some(removed))
+        let removed = inner.entries[entry_index].clone();
+        inner.entries[entry_index] = None;
+
+        Ok(Some(removed.unwrap()))
```

Comment:
```suggestion
        let removed = std::mem::take(&mut inner.entries[entry_index]);
        Ok(removed)
```
`std::mem` has a few tools to avoid clone